jlibra-core/src/main/java/dev/jlibra/admissioncontrol/transaction/Signature.java

```diff
@@ -39,4 +42,45 @@ public static Signature signTransaction(Transaction transaction, PrivateKey priv
                 .build();
     }
 
+    public static Signature addSignatureToMultiSignature(Signature signature, int index, Transaction transaction,
+            PrivateKey privateKey) {
+        Signature signatureToAdd = signTransaction(transaction, privateKey);
+
+        byte[] bitmap = signature.getSignature()
+                .subseq(signature.getSignature().toArray().length - BITMAP_LENGTH,
+                        BITMAP_LENGTH)
+                .toArray();
+
+        int bucket = index / 8;
+        int bucket_pos = index - (bucket * 8);
+        bitmap[bucket] |= 128 >> bucket_pos;
```

Comment:
I don't think there is a clearer way to implement this because at least I'm not aware of java classes that could be used to build the required bitmap.
I added some comments and refactored the code, I hope it is now easier to understand.